sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/KafkaPropertiesBeanPostProcessor.java

```diff
@@ -0,0 +1,91 @@
+    private static final String LOG_PROPERTIES_CONFIGURE = "Property %s will be configured as %s.";
+    private final AzureEventHubsKafkaProperties azureEventHubsKafkaProperties;
+
+    KafkaPropertiesBeanPostProcessor(AzureEventHubsKafkaProperties azureEventHubsKafkaProperties) {
+        this.azureEventHubsKafkaProperties = azureEventHubsKafkaProperties;
+    }
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        if (bean instanceof KafkaProperties) {
+            KafkaProperties kafkaProperties = (KafkaProperties) bean;
+            //only map to common properties instead of upper admin/consumer/producer properties given these are Spring Cloud Azure
+            //self-defined properties
+            mapAzurePropertiesToKafkaCommonProperties(kafkaProperties);
+            
+            configureOAuthPropertiesIfNeed(kafkaProperties.buildAdminProperties(), kafkaProperties.getAdmin().getProperties());
+            configureOAuthPropertiesIfNeed(kafkaProperties.buildConsumerProperties(), kafkaProperties.getConsumer().getProperties());
+            configureOAuthPropertiesIfNeed(kafkaProperties.buildProducerProperties(), kafkaProperties.getProducer().getProperties());
+        }
+        return bean;
+    }
+
+    private void configureOAuthPropertiesIfNeed(Map<String, Object> sourceProperties,
+                                                Map<String, String> propertiesToConfigure) {
+        if (ifSaslOAuthNeedConfigure(sourceProperties)) {
+            propertiesToConfigure.put(SECURITY_PROTOCOL_CONFIG, SECURITY_PROTOCOL_CONFIG_SASL);
+            LOGGER.debug(LOG_PROPERTIES_CONFIGURE, SECURITY_PROTOCOL_CONFIG, SECURITY_PROTOCOL_CONFIG_SASL);
+            propertiesToConfigure.put(SASL_MECHANISM, SASL_MECHANISM_OAUTH);
+            LOGGER.debug(LOG_PROPERTIES_CONFIGURE, SASL_MECHANISM, SASL_MECHANISM_OAUTH);
+            propertiesToConfigure.put(SASL_JAAS_CONFIG, SASL_JAAS_CONFIG_OAUTH);
+            LOGGER.debug(LOG_PROPERTIES_CONFIGURE, SASL_JAAS_CONFIG, SASL_JAAS_CONFIG_OAUTH);
+            propertiesToConfigure.put(SASL_LOGIN_CALLBACK_HANDLER_CLASS, SASL_LOGIN_CALLBACK_HANDLER_CLASS_OAUTH);
+            LOGGER.debug(LOG_PROPERTIES_CONFIGURE, SASL_LOGIN_CALLBACK_HANDLER_CLASS, SASL_LOGIN_CALLBACK_HANDLER_CLASS_OAUTH);
+        }
+    }
+
+    /**
+     * Detect whether we need to configure SASL/OAUTHBEARER properties for {@link KafkaProperties}. Will configure
+     * when the security protocol is not configured, or it's set as SASL_SSL with sasl mechanism as null or OAUTHBEAR.
+     *
+     * @param sourceProperties the source kafka properties for admin/consumer/producer to detect
+     * @return whether we need to configure with Spring Cloud Azure MSI support or not.
+     */
+    private boolean ifSaslOAuthNeedConfigure(Map<String, Object> sourceProperties) {
```

Comment:
how about "needConfigureSaslOAuth"?

Reply:
updated in https://github.com/Azure/azure-sdk-for-java/pull/29404